packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -249,10 +313,15 @@ export class ConnectionService {
           problemCode: ConnectionProblemReportReason.RequestProcessingError,
         })
       }
+      throw error
     }
 
     const connection = JsonTransformer.fromJSON(connectionJson, Connection)
-    await MessageValidator.validate(connection)
+    try {
+      await MessageValidator.validate(connection)
+    } catch (error) {
+      throw new Error(error)
+    }
```

Comment:
We should probably add this wrapper in the `MessageValidator` itself

Reply:
Agree. But probably in separate PR.